app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v3.rb

```diff
@@ -227,9 +227,24 @@ def method_missing(method_name, *args, &block)
       end
     end
 
-    def find_mac_address_on_network(nics, network, log)
+    def get_mac_address_of_nic_on_requested_vlan(args)
+      network = find_network_in_cluster(args[:value_of_vlan_option], args[:dest_cluster], args[:source_template])
+      return nil if network.nil?
+
+      find_mac_address_on_network(network, args[:destination])
+    end
+
+    def find_network_in_cluster(network_name, dest_cluster, source)
+      network = source.ext_management_system.ovirt_services.cluster_find_network_by_name(dest_cluster.ems_ref, network_name)
```

Comment:
1. I think its proper to put this method here since in v4 it is not used so it does not make sense to have it in network.rb (actually this is exactly the reason why we split both of them - in v3 we find network by name, in v4 we do not need it since we have the vnic_profile, you were the one who explained that to me...)
2. nics_for_vm does not use the ext_management_system, it make a call on the vm directly.  The cluster_find_network_by_name is used in other places, so I would rather not replacement it to use vm.